include/pros/aivision.hpp

```diff
@@ -0,0 +1,112 @@
+/**
+* \file pros/aivision.hpp
+* \ingroup cpp-aivision
+*
+* Contains prototypes for the VEX AI Vision Sensor-related functions in C++.
+*
+* This file should not be modified by users, since it gets replaced whenever
+* a kernel upgrade occurs.
+*
+* \copyright (c) 2017-2024, Purdue University ACM SIGBots.
+* All rights reserved.
+*
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*
+* \defgroup cpp-aivision AI Vision Sensor C++ API
+* \note Additional example code for this module can be found in its [Tutorial.](@ref aivision)
+*/
+
+#ifndef _PROS_AIVISION_HPP_
+#define _PROS_AIVISION_HPP_
+
+#include <cstdint>
+
+#include "pros/aivision.h"
+#include "pros/device.hpp"
+
+namespace pros {
+inline namespace v5 {
+
+enum class AivisionDetectType : uint8_t { color = (1 << 0), code = (1 << 1), object = (1 << 2), tag = (1 << 3) };
```

Comment:
be nice to have this on seperate lines for readability's sake:
```suggestion
enum class AivisionDetectType : uint8_t { 
    color = (1 << 0), 
    code = (1 << 1), 
    object = (1 << 2),
    tag = (1 << 3) 
};
```